File: hello-main.c
int my_printk(const char* str);

int init_module(void)
{
    my_printk("<1>Hello-main:init()\n");
    return 0;
}

void cleanup_module(void)
{
    my_printk("<1>Hello-main:exit() -- Goodbye\n");
}

File: hello-main.rs
extern {
    fn printk(str: &[u8]);
}

#[no_mangle]
pub fn init_module() -> isize {
    unsafe {
        printk("rust_linux: Hello Linux from Rust\n".as_bytes());
    }
    return 0;
}

#[no_mangle]
pub fn cleanup_module() {
    unsafe {
       printk("rust_linux: Goodby Linux from Rust\n".as_bytes());
    }
}
